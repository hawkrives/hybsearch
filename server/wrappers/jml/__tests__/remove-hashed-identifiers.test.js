/* eslint-env jest */

const revertHashedIdentifiers = require('../revert-hashed-identifiers')

describe('replaces the hashed identifiers with the unhashed ones', () => {
	let mapping = {
		blgcgdgcx1: 'Emydura_subglobosa__KC755190',
		ahjmjgejx1: 'Emydura_victoriae__KC755189',
		ahjmjgejx2: 'Emydura_victoriae__KC755188',
		defdnggjx1: 'Emydura_tanybaraga__KC755187',
		defdnggjx2: 'Emydura_tanybaraga__KC755186',
		blgcgdgcx2: 'Emydura_subglobosa__KC755185',
		blgcgdgcx3: 'Emydura_subglobosa__KC755184',
		gpnkepggx1: 'Emydura_macquarii__KC755183',
		blgcgdgcx4: 'Emydura_subglobosa__NC_026048',
		blgcgdgcx5: 'Emydura_subglobosa__KC692462',
	}

	test('in results', () => {
		let results = `Sp Comparison	seq1	seq2	Distance	Probability
gpnkepgg-blgcgdgc	gpnkepgg1	blgcgdgc1	0.00449985	0.0944528
gpnkepgg-blgcgdgc	gpnkepgg1	blgcgdgc2	0.00474308	0.0944528
gpnkepgg-blgcgdgc	gpnkepgg1	blgcgdgc3	0.00492551	0.0944528
gpnkepgg-ahjmjgej	gpnkepgg1	ahjmjgej1	0.00224992	0.0394803
gpnkepgg-ahjmjgej	gpnkepgg1	ahjmjgej2	0.00218912	0.0394803
gpnkepgg-defdnggj	gpnkepgg1	defdnggj1	0.00529036	0.0784608
gpnkepgg-defdnggj	gpnkepgg1	defdnggj2	0.00529036	0.0784608`
		let probabilities = ''
		let distributions = ''

		let actual = revertHashedIdentifiers({
			phylipMapping: mapping,
			results,
			probabilities,
			distributions,
		})
		expect(actual).toMatchSnapshot()
	})

	test('in distributions', () => {
		let distributions = `ahjmjgej-blgcgdgc	defdnggj-blgcgdgc	defdnggj-ahjmjgej	gpnkepgg-blgcgdgc	gpnkepgg-ahjmjgej	gpnkepgg-defdnggj
0	0	0	0	0	0
0	0	0	0	0	0
0	0	0	0	0	0
0	0	0	0	0	0`
		let results = ''
		let probabilities = ''

		let actual = revertHashedIdentifiers({
			phylipMapping: mapping,
			results,
			probabilities,
			distributions,
		})
		expect(actual).toMatchSnapshot()
	})

	test('in probabilities', () => {
		let probabilities = `Comparison	minDist	Probability
ahjmjgej-blgcgdgc	0.00462147	0.118441
defdnggj-blgcgdgc	0.000304044	0.130935
defdnggj-ahjmjgej	0.00504713	0.102449
gpnkepgg-blgcgdgc	0.00449985	0.0944528
gpnkepgg-ahjmjgej	0.00218912	0.0394803
gpnkepgg-defdnggj	0.00529036	0.0784608`
		let results = ''
		let distributions = ''

		let actual = revertHashedIdentifiers({
			phylipMapping: mapping,
			results,
			probabilities,
			distributions,
		})
		expect(actual).toMatchSnapshot()
	})
})
