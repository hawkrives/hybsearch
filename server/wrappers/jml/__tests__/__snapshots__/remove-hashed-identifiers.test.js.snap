// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaces the hashed identifiers with the unhashed ones in distributions 1`] = `
Object {
  "distributions": Array [
    Object {
      "Emydura macquarii-Emydura subglobosa": 0,
      "Emydura macquarii-Emydura tanybaraga": 0,
      "Emydura macquarii-Emydura victoriae": 0,
      "Emydura tanybaraga-Emydura subglobosa": 0,
      "Emydura tanybaraga-Emydura victoriae": 0,
      "Emydura victoriae-Emydura subglobosa": 0,
    },
    Object {
      "Emydura macquarii-Emydura subglobosa": 0,
      "Emydura macquarii-Emydura tanybaraga": 0,
      "Emydura macquarii-Emydura victoriae": 0,
      "Emydura tanybaraga-Emydura subglobosa": 0,
      "Emydura tanybaraga-Emydura victoriae": 0,
      "Emydura victoriae-Emydura subglobosa": 0,
    },
    Object {
      "Emydura macquarii-Emydura subglobosa": 0,
      "Emydura macquarii-Emydura tanybaraga": 0,
      "Emydura macquarii-Emydura victoriae": 0,
      "Emydura tanybaraga-Emydura subglobosa": 0,
      "Emydura tanybaraga-Emydura victoriae": 0,
      "Emydura victoriae-Emydura subglobosa": 0,
    },
    Object {
      "Emydura macquarii-Emydura subglobosa": 0,
      "Emydura macquarii-Emydura tanybaraga": 0,
      "Emydura macquarii-Emydura victoriae": 0,
      "Emydura tanybaraga-Emydura subglobosa": 0,
      "Emydura tanybaraga-Emydura victoriae": 0,
      "Emydura victoriae-Emydura subglobosa": 0,
    },
  ],
  "probabilities": Array [],
  "results": Array [],
}
`;

exports[`replaces the hashed identifiers with the unhashed ones in probabilities 1`] = `
Object {
  "distributions": Array [],
  "probabilities": Array [
    Object {
      "Comparison": "Emydura victoriae-Emydura subglobosa",
      "Probability": 0.118441,
      "minDist": 0.00462147,
    },
    Object {
      "Comparison": "Emydura tanybaraga-Emydura subglobosa",
      "Probability": 0.130935,
      "minDist": 0.000304044,
    },
    Object {
      "Comparison": "Emydura tanybaraga-Emydura victoriae",
      "Probability": 0.102449,
      "minDist": 0.00504713,
    },
    Object {
      "Comparison": "Emydura macquarii-Emydura subglobosa",
      "Probability": 0.0944528,
      "minDist": 0.00449985,
    },
    Object {
      "Comparison": "Emydura macquarii-Emydura victoriae",
      "Probability": 0.0394803,
      "minDist": 0.00218912,
    },
    Object {
      "Comparison": "Emydura macquarii-Emydura tanybaraga",
      "Probability": 0.0784608,
      "minDist": 0.00529036,
    },
  ],
  "results": Array [],
}
`;

exports[`replaces the hashed identifiers with the unhashed ones in results 1`] = `
Object {
  "distributions": Array [],
  "probabilities": Array [],
  "results": Array [
    Object {
      "Distance": 0.00449985,
      "Probability": 0.0944528,
      "Sp Comparison": "Emydura macquarii-Emydura subglobosa",
      "seq1": "Emydura macquarii (KC755183)",
      "seq2": "Emydura subglobosa (KC755190)",
    },
    Object {
      "Distance": 0.00474308,
      "Probability": 0.0944528,
      "Sp Comparison": "Emydura macquarii-Emydura subglobosa",
      "seq1": "Emydura macquarii (KC755183)",
      "seq2": "Emydura subglobosa (KC755185)",
    },
    Object {
      "Distance": 0.00492551,
      "Probability": 0.0944528,
      "Sp Comparison": "Emydura macquarii-Emydura subglobosa",
      "seq1": "Emydura macquarii (KC755183)",
      "seq2": "Emydura subglobosa (KC755184)",
    },
    Object {
      "Distance": 0.00224992,
      "Probability": 0.0394803,
      "Sp Comparison": "Emydura macquarii-Emydura victoriae",
      "seq1": "Emydura macquarii (KC755183)",
      "seq2": "Emydura victoriae (KC755189)",
    },
    Object {
      "Distance": 0.00218912,
      "Probability": 0.0394803,
      "Sp Comparison": "Emydura macquarii-Emydura victoriae",
      "seq1": "Emydura macquarii (KC755183)",
      "seq2": "Emydura victoriae (KC755188)",
    },
    Object {
      "Distance": 0.00529036,
      "Probability": 0.0784608,
      "Sp Comparison": "Emydura macquarii-Emydura tanybaraga",
      "seq1": "Emydura macquarii (KC755183)",
      "seq2": "Emydura tanybaraga (KC755187)",
    },
    Object {
      "Distance": 0.00529036,
      "Probability": 0.0784608,
      "Sp Comparison": "Emydura macquarii-Emydura tanybaraga",
      "seq1": "Emydura macquarii (KC755183)",
      "seq2": "Emydura tanybaraga (KC755186)",
    },
  ],
}
`;
